File: Data Structures & Algorithms/maximum-subarray/submission-0.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int m=nums[0];
        int sum=0;
        int l=0;
        for(int i=0;i<nums.size();i++){
            if(sum+nums[i]>=0){
                sum+=nums[i];
                m=max(sum,m);
            }
            else
                sum=0;
            m=max(m,nums[i]);
        }   
        return m; 
    }
};

File: Data Structures & Algorithms/maximum-product-subarray/submission-0.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        product=1
        m=nums[0]
        n=1
        for i in range(0,len(nums)):
            if product==0:
                product=nums[i]
            else:
                product=product*nums[i]
            if nums[i]<0 and n==1:
                n=product
                continue
            elif nums[i]==0:
                n=1
            if product<0:
                m=max(m,product/n)
            else:
                m=max(m,product)
        return int(m)